### PR TITLE
TAS-1113: Implement new database schema, migrate all services and implement event creation.

### PR DESCRIPTION
**Pull request by jarvistatch**

### CODE REVIEW

File: src/events/entities/event.entity.ts [hunk 1]
```diff
@@ -1,21 +1,41 @@
...
+  EventStatus as PrismaEventStatus,
+} from '@prisma/client';
+
+export type EventStatus = PrismaEventStatus;
 
 @ObjectType()
 export class Event {
-  @Field(() => Int)
-  id: number;
+  @Field(() => String)
```
Comment (Seesaem) [inline src/events/entities/event.entity.ts]: Use the Grapqhql Field type ID instead of String.

File: src/events/entities/event.entity.ts [hunk 2]
```diff
@@ -1,21 +1,41 @@
...
+
+  @Field(() => String)
+  name: string;
+
   @Field(() => String, { nullable: true })
-  name?: string;
+  description?: string;
+
+  @Field(() => String, { nullable: true })
+  status: EventStatus;
```
Comment (Seesaem) [inline src/events/entities/event.entity.ts:32]: status?: EventStatus;

File: src/locations/entities/location.entity.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { ObjectType, Field, Float } from '@nestjs/graphql';
+import { Location as LocationPrisma } from '@prisma/client';
+
+@ObjectType()
+export class Location {
+  @Field(() => String)
```
Comment (Seesaem) [inline src/locations/entities/location.entity.ts]: Field type should be ID, not String

File: src/loot-boxes/entities/loot-box.entity.ts [hunk 4]
```diff
@@ -2,57 +2,49 @@ import { Field, ObjectType, Int, GraphQLTimestamp, ID } from '@nestjs/graphql';
...
   openedBy?: User;
 
-  @Field(() => Int)
-  eventId: number;
+  eventId: string;
 
   @Field(() => Event, { nullable: true })
   event?: Event;
 
-  @Field(() => Int, { nullable: true })
```
Comment (Seesaem) [inline src/loot-boxes/entities/loot-box.entity.ts:37]: why remove the field type completly?

File: src/loot-boxes/entities/loot.entity.ts [hunk 5]
```diff
@@ -4,8 +4,8 @@ import { LootBox } from './loot-box.entity';
 
 @ObjectType()
 export class Loot {
-  @Field(() => Int)
-  id: number;
+  @Field(() => String)
```
Comment (Seesaem) [inline src/loot-boxes/entities/loot.entity.ts]: should be ID Field type

File: src/users/entities/user.entity.ts [hunk 6]
```diff
@@ -4,8 +4,8 @@ import { User as UserPrisma } from '@prisma/client';
 
 @ObjectType()
 export class User {
-  @Field(() => Int)
-  id: number;
+  @Field(() => String)
```
Comment (Seesaem) [inline src/users/entities/user.entity.ts]: should be ID not String

File: src/prisma/seeds/utils.ts [hunk 7]
```diff
@@ -25,3 +25,41 @@ export const seedTable = async (
...
+  const w = radiusInDegrees * Math.sqrt(u);
+  const t = 2 * Math.PI * v;
+  const deltaLat = w * Math.cos(t);
+  const deltaLon = (w * Math.sin(t)) / Math.cos(lat * (Math.PI / 180));
+
+  const newLat = lat + deltaLat;
+  const newLon = lon + deltaLon;
+
+  return { latitude: newLat, longitude: newLon };
+}
```
Comment (Seesaem) [inline src/prisma/seeds/utils.ts:47]: probably can do it with [fakerjs](https://fakerjs.dev/api/location.html#nearbygpscoordinate)

File: src/prisma/seeds/develop.seed.ts [hunk 8]
```diff
@@ -93,77 +22,77 @@ export const main = async () => {
...
-
-  return lootboxes;
-}
+  });
+};
 
 const createRandomUser = () => ({
   id: uniqueEnforcerId.enforce(() => {
-    return faker.number.int({ max: 1000000 });
+    return faker.database.mongodbObjectId();
```
Comment (Seesaem) [inline src/prisma/seeds/develop.seed.ts]: use the uuid of string of [fakerjs](https://fakerjs.dev/api/string.html#uuid)

**Review by Seesaem — CHANGES_REQUESTED** [review]

File: src/prisma/seeds/utils.ts (hunk 7 above)
Comment (jarvistatch) [reply src/prisma/seeds/utils.ts:47]: Nop, the point is to have random coordinate in a specific perimeter to be able to test the map. Faker returns locations that are too random and therefore hard to see in a zoomed view on the map.

**Pull request merged**